routing/router.go

```diff
@@ -1405,7 +1405,11 @@ func generateSphinxPacket(rt *route.Route, paymentHash []byte) ([]byte,
 
 	log.Tracef("Constructed per-hop payloads for payment_hash=%x: %v",
 		paymentHash[:], newLogClosure(func() string {
-			return spew.Sdump(sphinxPath[:sphinxPath.TrueRouteLength()])
+			path := sphinxPath[:sphinxPath.TrueRouteLength()]
+			for _, hop := range path {
+				hop.NodePub.Curve = nil
```

Comment:
are we sure this won't crash if we try to use these pubkeys later on? I saw such a crash that was fixed by https://github.com/lightningnetwork/lnd/pull/3065

Reply:
should be fine, the only operations done on these pubkeys afaict is `ScalarMult` and `ScalarBaseMult` when computing shared secrets. lightning-onion calls these methods directly on `btcec.S256()` so should be okay

Reply:
i ran all of the tests with `nolog trace` and nothing crashed